### PR TITLE
Make addgamerule command process only valid game rules

### LINKED ISSUE #29905: addgamerule shows admin message even if the rule doesnt exist

### PR DESCRIPTION
**Pull request by Winkarst-cpu**
## About the PR

Now ``addgamerule`` command processes only valid game rules.

## Why / Balance

Fixes #29905.

## Technical details

``addgamerule`` command now checks if game rule passed as an argument is valid.

## Media

https://github.com/space-wizards/space-station-14/assets/74284083/42523578-10e2-45da-8574-3509d4ba6593

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- fix: Now addgamerule command processes only valid game rules.

### CODE REVIEW

**Review by PJB3005 — CHANGES_REQUESTED** [review]
This should print an explicit error to the console so that the admin in question knows they fucked up.

**Comment by Winkarst-cpu** [comment]
> This should print an explicit error to the console so that the admin in question knows they fucked up.

изображение

**Review by PJB3005 — APPROVED** [review]

File: Content.Server/GameTicking/GameTicker.GameRule.cs [hunk 1]
```diff
@@ -324,6 +324,13 @@ private void AddGameRuleCommand(IConsoleShell shell, string argstr, string[] arg
 
         foreach (var rule in args)
         {
+            if (!_prototypeManager.HasIndex(rule))
+            {
+                shell.WriteError($"Invalid game rule {rule} was passed.");
```
Comment (Tayrtahn) [inline Content.Server/GameTicking/GameTicker.GameRule.cs:329]:
```suggestion
                shell.WriteError($"Invalid game rule {rule} was skipped.");
```

**Comment by Winkarst-cpu** [comment]
Failed check is unrelated to the PR...

**Pull request merged**